filewatcher/filewatcher.go

```diff
@@ -379,3 +426,19 @@ func (fw *MockFileWatcher) Get() interface{} {
 func (fw *MockFileWatcher) Stop() {}
 
 var _ FileWatcher = (*MockFileWatcher)(nil)
+
+// DummyReadCloser is a mock struct used to hold the path for directory watchers
+type DummyReadCloser struct {
+	io.Reader
+	io.Closer
```

Comment:
[the `Parser` interface only requires `io.Reader`](https://pkg.go.dev/github.com/reddit/baseplate.go/filewatcher#Parser), so there's no need to add dummy closer here.

Also please add blank lines between embedded and non-embedded fields.

Reply:
that is true about the parser but the limitopen module returns a ReadCloser, so many parts of the code treat it as such, take the watcherloop for example.

I will add the space for the embedded field.